apps/google-analytics-4/lambda/src/services/dynamoDbService.ts

```diff
@@ -0,0 +1,63 @@
+import { DynamoDBClient } from '@aws-sdk/client-dynamodb';
+import { DynamoDBDocumentClient, GetCommand, PutCommand } from '@aws-sdk/lib-dynamodb';
+import { config } from '../config';
+import { decryptSharedCredentials, encryptSharedCredentials } from './dynamoDbUtils';
+
+export interface SharedCredentialsInput {
+  sharedCredentialsId: string;
+  serviceKey?: object;
+}
+
+export class DynamoDBService {
+  private dynamoDBDocumentClient: DynamoDBDocumentClient;
+  private tableName: string;
+
+  constructor(
+    client = new DynamoDBClient({
+      region: config.awsRegion,
+      endpoint: config.dynamoDbEndpoint,
+    })
+  ) {
+    this.dynamoDBDocumentClient = DynamoDBDocumentClient.from(client);
+    this.tableName = config.dynamoDbTableName;
+  }
+
+  async getSharedCredentials(input: SharedCredentialsInput) {
```

Comment:
Take this as a follow up PR but I'm not seeing the value of sharing `input` between these two methods.

This is a pretty common pattern for getters and setters where you have:

get(key)
set(key, value)

And i think ideally what you're doing is much the same:

getSharedCredentials(sharedCredentialsId)
saveSharedCredentials(sharedCredentialsId, sharedCredentials)

Reply:
You are spot on, I will add this cleanup in the client side PR coming tomorrow 👍